docs/Tutorials/EventsAndTriggers.md

```diff
@@ -0,0 +1,72 @@
+effects on the flow of the algorithm, but is primarily used for
+observations and minor adjustments.
+
+### What is an event?
+
+An \ref Event "event" is an input-file-creatable object that performs
+some task, frequently an observation.  With the exception of the \ref
+Events::Completion "Completion" event, they cannot have direct effects
+on the algorithm, although they can have indirect effects through
+sending messages.  (An example of such an event is \ref
+Events::ChangeSlabSize "ChangeSlabSize".)  In most cases, however, their
+effects on program execution are ephemeral, causing some data to be
+written to a file or standard output but not changing anything about
+the simulation state.
+
+### What is a trigger?
+
+A \ref Trigger "trigger" is an input-file-creatable object that
+controls when events are run.  They are checked periodically, once per
+slab in an evolution executable, and once per iteration in an elliptic
+solve.  At each check, a trigger may fire, causing its associated
+events to run, or not.  Triggers must always be written to be
+independent of element-specific state such as the local values of the
+system variables, so that a trigger will give a consistent result over
+the entire domain at each check.
+
+### What is a dense trigger?
+
+A \ref DenseTrigger "dense trigger" is similar to a trigger, but uses
+time-stepper dense output to provide more precise control of the time
+that it fires.  As a time-related feature, dense triggers are only
+applicable to evolution executables.  Dense triggers fire at a precise
+set of times, independent of the time-step or slab state, by
+interpolating the evolved variables to the requested time before
+running the associated events.  As with normal triggers, dense
+triggers are required to fire consistently over the entire domain.
+
+### What's with the weird input file syntax?
+
+The `%EventsAndTriggers` (and `%EventsAndDenseTriggers`) sections of
+the input file indicate structure using a collection of `?`, `:`, and
+`-` characters.  These are standard, if perhaps somewhat obscure, YAML
+syntax.  They are defining a map from triggers to lists of events.
+The `?:` syntax defines an expanded key-value pair, which allows the
+key to be a complicated type, unlike the common `Key: Value` syntax.
+The `-` is the expanded form of a list, which allows entries to have
+complicated types.  Frequently only one event is associated with a
+trigger, so there will only be a single `-` after the `:`, but
+additional events introduced by more `-` characters can appear
+afterwards:
+
+\snippet PlaneWave1DEventsAndTriggersExample.yaml multiple_events
+
+If an event and trigger both took no options, the syntax
+
+\snippet PlaneWave1DEventsAndTriggersExample.yaml no_arguments
+
+could be written in the more familiar form
+
+\snippet PlaneWave1DEventsAndTriggersExample.yaml no_arguments_single_line
+
+but we recommend always using the expanded form for consistency.
```

Comment:
If you don't want users to use this format for triggers/events with no options, then don't put it in. It'll only confuse them more. What about this for this part:
```suggestion
If an event and trigger both took no options, the recommended syntax is still

\snippet PlaneWave1DEventsAndTriggersExample.yaml no_arguments
```

Reply:
This was more of a nitpick so I think it's ok as is.